packages/search/src/features/registration/marriage/service.ts

```diff
@@ -161,41 +160,37 @@ async function indexDeclaration(
     operationHistories: (await getStatus(compositionId)) as IOperationHistory[]
   }
 
-  await createIndexBody(body, composition, authHeader, bundleEntries)
+  await createIndexBody(body, composition, authHeader, bundle)
   await indexComposition(compositionId, body, client)
 }
 
 async function createIndexBody(
   body: IMarriageCompositionBody,
-  composition: fhir.Composition,
+  composition: SavedComposition,
   authHeader: string,
-  bundleEntries?: fhir.BundleEntry[]
+  bundle: SavedBundle
 ) {
-  await createBrideIndex(body, composition, bundleEntries)
-  await createGroomIndex(body, composition, bundleEntries)
-  createWitnessOneIndex(body, composition, bundleEntries)
-  createWitnessTwoIndex(body, composition, bundleEntries)
-  await createDeclarationIndex(body, composition, bundleEntries)
-  const task = findTask(bundleEntries)
-  await createStatusHistory(body, task, authHeader)
+  await createBrideIndex(body, composition, bundle)
+  await createGroomIndex(body, composition, bundle)
+  createWitnessOneIndex(body, composition, bundle)
+  createWitnessTwoIndex(body, composition, bundle)
+  await createDeclarationIndex(body, composition, bundle)
+  const task = getTaskFromSavedBundle(bundle)
+  await createStatusHistory(body, task, authHeader, bundle)
 }
 
 async function createBrideIndex(
   body: IMarriageCompositionBody,
-  composition: fhir.Composition,
-  bundleEntries?: fhir.BundleEntry[]
+  composition: SavedComposition,
+  bundle: SavedBundle
 ) {
-  const bride = findEntry(
-    BRIDE_CODE,
-    composition,
-    bundleEntries
-  ) as fhir.Patient
+  const bride = findEntry(BRIDE_CODE, composition, bundle) as unknown as Patient
```

Comment:
To keep the return type of `findEntry` consistent, it would be better to change it to `as Patient | undefined` from `as Patient`